### DISCUSSION TITLE
2023-06-30 - Data Engineering #3 - Data Warehouses - Chewing the Fat 📈

### DISCUSSION THREAD

**Discussion by lukecookssw** [SSWConsulting/SSW.Rules.Content#5894] [Chewing the Fat]
Hey data engineers!

Let's talk about [this rule](https://www.ssw.com.au/rules/do-you-use-a-data-warehouse/)

**Comment by camillars** [comment]
Data warehouses: where data magic happens, insights shine, and decisions thrive. 🤖🦸😎

**Comment by wicksipedia** [comment]
Oh, absolutely! I just love manually sifting through countless spreadsheets, juggling multiple databases, and desperately trying to piece together inconsistent data. Who needs a fancy Data Warehouse when chaos and confusion are so much more exciting?

![image](https://github.com/SSWConsulting/SSW.Rules.Content/assets/600044/7a1ba286-f1e1-4472-ae10-5d857fcaab52)

**Comment by jaydenalchin** [comment]
![image](https://github.com/SSWConsulting/SSW.Rules.Content/assets/75593567/98dda486-c98d-483e-9563-e09c9fea13ea)

**Comment by vladislav-kir** [comment]
It was a nice CTF for explaining the use cases for Data Warehouses.

**Comment by Marxoz** [comment]
There is just something about working on a client with a proper DW in place. I love working with data, but it has to be clean(ed).

Both the rules and the video is good (5⭐️), but it might be worth diving deeper into Facts, dimensions, measures etc in another rule / CTF

**Comment by jeoffreyfischer** [comment]
Good to dig into those different types of data storage concepts, I never heard about those before and it's definitely very interesting!

**Comment by matt-goldman** [comment]
Just seems like a fancy name for a database that aggregates data from different sources for reporting purposes 🤷

**Reply by jakebayliss** [reply]
Gourmet Data Restaurant

**Reply by ozdemir-mehmet** [reply]
If it was a fancy name it would be Data HyperWarehousePro!

**Reply by calumjs** [reply]
Gourmet Data Restaurant with a Michelin Star Schema

**Reply by ozdemir-mehmet** [reply]
How many hats?

**Reply by matt-goldman** [reply]
SugarWarehouse

**Reply by ozdemir-mehmet** [reply]
MrWhippyWarehouse

**Reply by tkapa** [reply]
Super Data Congregation Plus

**Reply by brydeno** [reply]
Just wait for Microsoft Silk (the next iteration of Fabric)

**Comment by lukecookssw** [comment]
A valuable point that @ozdemir-mehmet raised is: a proper Data Warehouse will often take years + $ millions to architect and implement. Warehouses are concerned with data measured in terabytes (or more).

**Comment by ncn-ssw** [comment]
I wasn't aware of the Star and Snowflake schema concepts before this CTF.

**Reply by ozdemir-mehmet** [reply]
Here's some more you might want to look at: https://www.fivetran.com/learn/what-is-a-database-schema

**Comment by MarcusIrmscherSSW** [comment]
Nice explanation, even I could get my head around it. 😀

**Comment by Levijj22** [comment]
Clear and concise video, great rule!

**Comment by ozdemir-mehmet** [comment]
The next thing to talk about is standardised Data Models: https://learn.microsoft.com/en-us/azure/cloud-adoption-framework/scenarios/cloud-scale-analytics/architectures/common-industry-data-models

No need to keep reinventing this stuff.

**Comment by calumjs** [comment]
Hey there! I've given your article a read and I think it's a fantastic primer on data warehouses - really detailed and informative. Just got a couple of suggestions to toss your way:

Your introduction is solid, but how about starting with a question or a problem that data warehouses can solve? This way, you immediately engage readers who might be facing that issue.

The YouTube link you provided is great! Just to make sure your readers click on it, maybe you could add a little more context or a teaser about what insights they could gain from the video.

The 'When to use a Data Warehouse' and 'When not to use a Data Warehouse' sections are great and clear. However, some visuals, like infographics, could help break up the text and make this information even more digestible.

In your 'How to create a Data Warehouse' section, you've done a good job outlining the key steps. It could be helpful to add a step about maintaining and evolving the data warehouse once it's built - things don't just stop once the warehouse is up and running!

Your list of popular Data Warehouse products is helpful. Consider giving a brief one-sentence summary for each, to help your readers make an even more informed decision.

Overall, this is super solid work! Just a little more tweaking and it'll be perfect. Keep it up!

🤖

**Comment by AshAnil** [comment]
Nice video and rule 👍 Well explained

**Comment by sethdaily** [comment]
After reading this guideline, I now understand that a Data Warehouse is a central repository for data, used for complex reporting and supporting decisions. I learned it's not suitable for real-time transactions or small-scale data storage. I know the process of creating one involves defining my goals, assessing my data, and strategizing. I'm also aware of several popular Data Warehouse products, and I'll choose based on my specific needs.
🤖⭐ 

Digestible even for someone with little data knowledge. Interesting!

**Comment by apforsyth** [comment]
great explanation.  I always love plain English.

**Comment by AlexRothwell** [comment]
It's great to have this concept explained in such detail, it often gets thrown around essentially as a buzzword. There is a little duplication would could be reduced but otherwise reads clearly

**Comment by bahjatalaadelssw** [comment]
Planning is important and to architect a scalable platform that won't cause thousands in hours to sort through data

**Comment by Ben0189** [comment]
Great video to explain data warehouse☺️

**Comment by SamBWagner** [comment]
first things first, lol on the acronyms haha.
 
![image](https://github.com/SSWConsulting/SSW.Rules.Content/assets/66365977/33a58adf-da8d-427e-9a79-38cb0c852aee)

I found the rule overly declarative and didn't give me a great understanding of Data Warehouses beyond what they were. I would've loved a few examples. :)

Otherwise great! Data is huge so having rules/ctfs like these is super important for us as a consultancy. 

Erryone got data yo 😎

**Comment by landonmaxwell** [comment]
I like how data warehouses centralize and organize information, empowering informed decisions, strategic insights, and improved efficiency.

**Comment by amankumarrr** [comment]
I never used it but would like to try it if I get a chance to work on big data.

**Comment by jomelsontanco** [comment]
Feedback:
My initial thoughts on this is that it might be misleading to teach this as a "new" concept. I fee like data engineering is just a term invented by business students who suddenly have to contend with an existing concept in existence for decades just because it has suddenly became profitable to do so. The signs are there. The language of data engeering is infected with business jargon and sales talk.

Potential Action:
Maybe it would be more useful in tracing back the history of how reporting has evolved. Personally curious on why "Data Warehouses" is a thing. Has reporting requirements really changed over the years? If yes, how and in what why? 

Maybe differentiating the "why" of data reporting (because users need reports) and the "how" (how it is implemented, the spread of readily accessible new technologies) would better bring home this point.

**Comment by 2fernandez** [comment]
I like that the rule is well written.
On another note, I am selling t-shirts this week for $15 for women's tops and $20 for men's tops. 

Is anyone interested?

![Screenshot 2023-06-30 at 9 57 12 am](https://github.com/SSWConsulting/SSW.Rules.Content/assets/27670973/0d0d1eb7-09a8-4ea6-b871-c0880d5c4c21)

**Reply by tkapa** [reply]
How about $2 and free lunch? (Provided by SSW)

**Reply by jernejk** [reply]
That could be an official intern/junior dev SSW T-shirt. 😂

**Reply by jaydenalchin** [reply]
Does it come in dark mode?

**Reply by 2fernandez** [reply]
![Screenshot 2023-06-30 at 10 45 19 am](https://github.com/SSWConsulting/SSW.Rules.Content/assets/27670973/e2104f2e-b8dd-4429-a09f-cd2d1af7d145)

@jernejk 
![Screenshot 2023-06-30 at 10 47 09 am](https://github.com/SSWConsulting/SSW.Rules.Content/assets/27670973/533f0231-c3a6-42bb-9a6c-8b246cd6c752)

**Reply by jernejk** [reply]
👉 AI Hack Day
![ShutUpAndTakeMyMoneyFuturamaGIF](https://github.com/SSWConsulting/SSW.Rules.Content/assets/5943653/ad90fd36-509e-49cb-b34e-3001f75634e9)

**Comment by tkapa** [comment]
Good to have a rule that can explain this concept that maybe we've heard about but haven't engaged with unless we're in the data side.

I personally think that the "Why" and the "How" may be better separated into 2 different rules, so that they aren't as long.

Warehouses seem cool, but I've never had to work with a data set large enough to worry about needing one.

**Comment by jakebayliss** [comment]
We should use a data warehouse for our reporting :)

**Comment by jackreimers** [comment]
I haven't ever encountered a scenario where I think a data warehouse would fit nor have I used one, but it would be interesting to work with one to learn more about their practical application

**Comment by christoment** [comment]
Keen to see how data warehouse works in a project :)

**Comment by Harry-Ross** [comment]
I thought the video was a great introduction to data warehouses, with all of the concepts explained really concisely. I haven't yet had a chance to use them, but they seem like a great way to aggregate clean data.

**Comment by KahaMason** [comment]
Great to see we now have rules for Business Intelligence & Data Warehouses. It's definitely a great area to dive deeper into. 😄

**Comment by JackDevAU** [comment]
✅Always good to learn new things! 
✅Rule covers some good content + the rule reads nicely! 

I'm sure we will be using Data warehouses more and more.

**Comment by cvionnet** [comment]
Nice rule and video.
Keen to learn more about DAX syntax.

**Comment by jernejk** [comment]
I wonder if using Microsoft Fabrick allows you to offload many decisions to MS instead.

**Reply by AntPolkanov** [reply]
What sort of decisions?🤔 Can you please give an example?

**Comment by tombui99** [comment]
Data warehouse is definitely a field that I'm interested in since it was one of my favourite subjects back in uni. While I do enjoy analysing and visualising the data on software such as Tableau, cleansing it can be quite a pain.

**Comment by Abf09** [comment]
Interested in getting more hands on industry experience with data warehousing. Curious to integrate in a project. Video seemed brief and easy to understand.

**Comment by AntPolkanov** [comment]
Nice CTF! 
I played with Data Warehousing and ETL tools a bit when I was at uni. Really liked the idea that you can separate transactional and analytical databases. Locks caused by complex reports gave me a lot of pain in the beginning of my career🥴

**Comment by steven0x51** [comment]
The video was a good introduction to Data Warehouse. 
Great rules.

**Comment by PennyWalker** [comment]
I'm really interested in all of the different ways to store data and the best practices around it, so this was really cool to know. I've had a few clients with big data projects, so it was useful to learn more about what other clients might need in the future.

**Comment by leahy268** [comment]
When I originally started with data I had no ability to create a datawarehouse.  To do so I would have had to prove it was required, get 3 quotes and so forth.  I instead did all of the ETL stuff directly in PowerBI from mulitple sources.  That was a very painful and time consuming process.

**Comment by MattParkerDev** [comment]
I have yet to use Data Warehouses, but will be interested to use them and explore their use cases

**Comment by drwharris** [comment]
One of those things that you need to suffer the pain to learn the lessons.

**Comment by UlyssesMaclaren** [comment]
Does Power BI have its own Data Warehouse? feels like it but maybe that's called something else

**Reply by leahy268** [reply]
No not really,  It has it's own area where data can be stored and manipulated but not to the extent or speed of a data warehouse.

**Comment by zacharykeeping** [comment]
Good video and rule. Interested to work with data warehouses in a project.

**Comment by piers-sinclair** [comment]
Data Warehouses are an important topic, but it's really difficult to illustrate how it practically differentiates from other solutions.

**Comment by Jord-Gui** [comment]
Nice video on data warehousing

**Comment by tomek-i** [comment]
I knew the concept of the Star Schema but didn't know the terminology behind it.
Thats the [snowflake](https://www.msn.com/en-au/money/watchlist?tab=Related&id=bvlqa2&ocid=ansMSNMoney11&duration=1D&relatedQuoteId=bvlqa2&relatedSource=MlAl&src=b_secdans) i know.

**Comment by Dhruv-0987** [comment]
Yeah data warehousing is a popular topic. I personally don't like unstructured data but its worth knowing the architecture behind it.

**Comment by jimmy1992abc** [comment]
If the Data Warehouse cannot deal with Real-time Transactional Processing, why not separate the Data Warehouse into the core and skin, where the skin can quickly respond to the real-time data and merge into the core in bulk?

**Comment by Hona** [comment]
I hadn't heard the terms dimension & fact but they link up to intuitive DB design

**Comment by AttackOnMorty** [comment]
It's important to ensure the quality of the data.

**Comment by jimmidier** [comment]
Great video! I'd love to see an example of integrating Azure Log Analytics into DW to see how we can make use of time series data which can easily grow to a large volume.

**Comment by Aibono1225** [comment]
I like the example given in the rule. It clarifies the use case of the data warehouse for me.

**Comment by Berber-Mama** [comment]
The video is great and very friendly for a non-dev like me to understand.

**Comment by AnkyChe** [comment]
Nice video and rule. The rule shows clearly when to use a data warehouse and how.

**Comment by sylhuang** [comment]
Nice intro to Data Warehouse and suitable use cases. Would be nice to see some demo project/playground to get hands on.

**Comment by KristenHu** [comment]
Nice video, never used Data Warehouse before, it's a great introduction to me.

**Comment by jerryluossw** [comment]
Many years ago: Data Warehouse + BI = Data Middle Platform
Now: Data Warehouse + BI + AI = Digital Auxiliary Governance
The Data Warehouse is foundation.

**Comment by tino-liu** [comment]
"Single source of truth"... Where did I hear that before? I think it's a sacred book.
Seems like data warehouse is what search engines run on...gonna read up more on it. Really cool rule!

**Comment by JaneYangssw** [comment]
Nice video, I'll learn more about Data Warehouse in the future😊.

**Comment by johnxu16** [comment]
I Heard of data warehouses a couple of years ago, but I never used any of them though. Building up a HA and scalable data warehouse needs a broad range of technologies and a theoretical understanding of how they work together.

**Comment by zl24622** [comment]
Great video. Simple, clean and informative. Learned some database basic concepts. 
By reading the rule, understood having a  data warehouse is not a simple quick job. Need experienced engineer, careful planning time and $.
Wondering if Agile Methodology can apply to warehouse development.

**Comment by ozair-ashfaque** [comment]
I did not know about the Star and Snowflake schema which I learned from this CTF

**Comment by william-liebenberg** [comment]
It's been a while since I had to look at data warehousing. The last time was around 2018/2019 while helping a client in the early days of spec'ing out and building some POCs for their data processing pipeline. 

We settled on using the Lambda architecture (as opposed to the simpler Kappa architecture) to have a real-time (non-curated) data stream and a slow (curated) historical data stream.

The real-time data stream ended up on a CosmosDB store and visualised with PowerBI. We built a bunch of Azure Functions that were triggered from their ESB and directly from their workflow platform. 

The same data was the pushed into the slow (curated) batch stream and to be processed using either Hadoop or Spark which and ended up on their centralized data warehouse. 

Big data projects are fun :) lots of technology choices out there.

**Comment by PothieuG** [comment]
Thanks to the chewing the fat, I learned about data Warehouse, star and cornflake schemas, and Fact and Dimension !
It was very interesting :)

**Comment by BrookJeynes** [comment]
Data warehouses can be useful but must be architected correctly otherwise it could cause big costs and problems if it must be changed later down the line

**Comment by TobyChurches** [comment]
I haven't used a Data Warehouse before, but they seem useful. Both the video and rule were easy to follow 🙂

**Comment by danielmackay** [comment]
I've never had to work with a Data Warehouse before, but I finally have a good understanding of where they fit into the ecosystem of data stores.

**Comment by stephanfakossw** [comment]
Easy to understand video, neat and concise rule.

**Comment by cameronshawssw** [comment]
Curious why an article about Data Warehousing makes its way into CTF?

**Comment by Khaled-Albahsh** [comment]
Data warehouse are cool, don't be a hater. be cool 😎

**Comment by JackDuan123** [comment]
It has always been great to have a pipeline tool for data engineer to use like Data Warehouse.
I would love to give it a try when I got a client that needs sort of data intergration and keen to build a ML project

**Comment by NadeekaJK** [comment]
A good simple topic for CTF!

**Comment by leehawkinsssw** [comment]
Data warehouses have been around for a long time now but one of the same key challenges persists - the quality of the data. The cleansing/transform piece discussed in the rule and video can be incredibly challenging for data entering from diverse sources and is a task not to be underestimated if the analytics from the data warehouse are really to have any meaning.

**Comment by JeanThirion** [comment]
Barely touched data warehousing... BUT WE NEED IT AT SSW ! Let's do our PowerBI Reporting from a Data Warehouse !

**Comment by chrisschultzssw** [comment]
Good introduction to data warehouses... plus I learned what a snowflake schema is.

**Comment by KostaMadorsky** [comment]
Does Power BI dataset count as a warehouse? 🙂
